packages/default-storage/android/config.gradle

```
@@ -0,0 +1,106 @@
+import java.nio.file.Paths
+
+def DEFAULT_KOTLIN_VERSION = "1.9.20"
+def DEFAULT_ROOM_VERSION = "2.4.3"
+
+project.ext.AsyncStorageConfig = [
+        kotlinVersion           : getKotlinVersion(),
+        kspVersion              : getKspVersion(kotlinVersion),
+        roomVersion             : getPropertyOfDefault('AsyncStorage_next_roomVersion', DEFAULT_ROOM_VERSION),
+        minSdkVersion           : safeExtGet('minSdkVersion', 23),
+        targetSdkVersion        : safeExtGet('targetSdkVersion', 32),
+        compileSdkVersion       : safeExtGet('compileSdkVersion', 32),
+        useNextStorage          : getFlagOrDefault("AsyncStorage_useNextStorage", false),
+        databaseSizeMB          : getDatabaseSize(),
+        isNewArchitectureEnabled: isNewArchitectureEnabled(),
+        useDedicatedExecutor    : getFlagOrDefault('AsyncStorage_dedicatedExecutor', false),
+]
+
+
+def getKotlinVersion() {
+    return rootProject.ext.has('kotlinVersion')
+            ? rootProject.ext['kotlinVersion']
+            : rootProject.hasProperty('AsyncStorage_kotlinVersion')
+            ? rootProject.properties['AsyncStorage_kotlinVersion']
+            : DEFAULT_KOTLIN_VERSION
+}
+
+def isNewArchitectureEnabled() {
+    // To opt-in for the New Architecture, you can either:
+    // - Set `newArchEnabled` to true inside the `gradle.properties` file
+    // - Invoke gradle with `-newArchEnabled=true`
+    // - Set an environment variable `ORG_GRADLE_PROJECT_newArchEnabled=true`
+    return project.hasProperty("newArchEnabled") && project.newArchEnabled == "true"
+}
+
+String getKspVersion(String kotlinVersion) {
+
+    String overriddenKspVersion = getPropertyOfDefault("AsyncStorage_next_kspVersion", null)
+    if (overriddenKspVersion != null) {
+        return overriddenKspVersion
+    }
+    // https://github.com/google/ksp/releases
+    return [
+            "1.9.20-1.0.14",
+            "1.9.10-1.0.13",
+            "1.9.0-1.0.13",
+            "1.8.22-1.0.11",
+            "1.8.21-1.0.11",
+            "1.8.20-1.0.11",
+            "1.8.10-1.0.9",
+            "1.8.0-1.0.9",
+            "1.7.22-1.0.8",
+            "1.7.21-1.0.8",
+            "1.7.20-1.0.8",
+            "1.7.10-1.0.6",
+            "1.7.0-1.0.6",
+            "1.6.21-1.0.6",
+            "1.6.20-1.0.5",
+            "1.6.10-1.0.4",
+            "1.6.0-1.0.2",
+            "1.5.31-1.0.1",
+            "1.5.30-1.0.0",
+    ].find { it.startsWith(kotlinVersion) }
```

We should have a fallback in case a newer version of Kotlin is used:
```suggestion
    def availableVersions = [
            "1.9.20-1.0.14",
            "1.9.10-1.0.13",
            "1.9.0-1.0.13",
            "1.8.22-1.0.11",
            "1.8.21-1.0.11",
            "1.8.20-1.0.11",
            "1.8.10-1.0.9",
            "1.8.0-1.0.9",
            "1.7.22-1.0.8",
            "1.7.21-1.0.8",
            "1.7.20-1.0.8",
            "1.7.10-1.0.6",
            "1.7.0-1.0.6",
            "1.6.21-1.0.6",
            "1.6.20-1.0.5",
            "1.6.10-1.0.4",
            "1.6.0-1.0.2",
            "1.5.31-1.0.1",
            "1.5.30-1.0.0",
    ]
    def version = availableVersions.find { it.startsWith(kotlinVersion) }
    return version ?: availableVersions[0]
```

---

I like fallback approach 👍